### ISSUE TITLE
Regression for date of birth formatting for English (United Kingdom)

### ISSUE THREAD

**Issue by MikeMcC399** [corona-warn-app/cwa-app-android#4112]
## Avoid duplicates
* [X] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [X] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [X] Bug is not already reported in another issue
Follows on from https://github.com/corona-warn-app/cwa-app-android/issues/4014 (EXPOSUREAPP-9297).

## Technical details

- Device name: Google Pixel 3a emulator
- Android version: 11
- First reported in App version: 2.10.1
- Confirmed still present in App version 2.19.2

## Describe the bug

If the device locale is set to English (United Kingdom) the date of birth shown in the "Born" field in a certificate displays with format `mm/dd/yyyy` which does not follow the locale standard of `dd/mm/yyyy` for English (United Kingdom).

(In version CWA 2.9.0 with locale set to English (United Kingdom) the date of birth was displayed as `dd.mm.yyyy`, which uses the correct order of day, month and year, although the delimiter "." is not normally used in UK. The delimiter used in the United Kingdom is commonly "/".)

## Steps to reproduce the issue

Locale set to English (United Kingdom)

On physical device:

1. Tap "Scan" button
2. Scan 2/2 vaccination certificate
3. Tap back arrow
4. Tap "Certificates"
5. QR code
6. Scroll down to "Born"
7. Note that date of birth is wrongly formatted as `mm/dd/yyyy`

Alternatively on Pixel 3a API 30 (Android 11) emulator

1. Settings > System > Languages and input > Languages<br>set to English (United Kingdom)
2. Settings > System > Date & time<br>disable "Use locale default"/ disable "Use 24-hour format"
3. Settings > Display > Advanced > Font Size, set Small
4. In Android Studio Build > Select `BuildVariant deviceForTestersDebug`
5. run `PersonDetailsFragmentTest.kt` on emulator
6. View > Tools Windows > Device File Explorer
7. Go to `/data/data/de.rki.coronawarnapp.test/screenshots`
8. View screenshots and check the formatting of all dates in PersonDetailsFragment*.png

See here for example `PersonDetailsFragment_booster_2.png`

![PersonDetailsFragment_booster_2-2 11 RC0-EN_GB](https://user-images.githubusercontent.com/66998419/134372390-c8cdcb81-2be6-4f52-a7d9-a44dd0353733.png)

## Expected behavior

If the device locale is set to English (United Kingdom) then a date of birth displayed in a numerical form should start with day of the month followed by month then year in that order as in `dd/mm/yyyy`.

## Possible Fix

Revise `formatBirthDate` in `Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/items/CwaUserCard.kt` to take care of device's chosen locale.

The delimiter "/" cannot be used to infer what order the fields `dd`, `mm` and `yyyy` should be displayed.

## Additional information

PR https://github.com/corona-warn-app/cwa-app-android/pull/4036 changed the date of birth formatting in certificates for [release/2.10.x](https://github.com/corona-warn-app/cwa-app-android/tree/release/2.10.x).

---
Internal Tracking ID: [EXPOSUREAPP-9595](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9595)

**Comment by MikeMcC399** [comment]
This should probably be looked at by @marcauberer.

**Comment by dsarkar** [comment]
Internal Tracking ID: [EXPOSUREAPP-9595](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9595)

**Comment by MikeMcC399** [comment]
According to https://github.com/ehn-dcc-development/ehn-dcc-schema/blob/release/1.3.0/DCC.combined-schema.json date of birth can be a full date, a year and a month, just a year or even blank:

```
    "dob": {
      "title": "Date of birth",
      "description": "Date of Birth of the person addressed in the DCC. ISO 8601 date format restricted to range 1900-2099 or empty",
      "type": "string",
      "pattern": "^((19|20)\\d\\d(-\\d\\d){0,2}){0,1}$",
      "examples": [
        "1979-04-14",
        "1950",
        "1901-08",
        ""
      ]
    }
```

Perhaps it would be better to align the display of dates of birth related to certificates in the app to the way they are displayed on the printed certificate `yyyy-mm-dd` independently of any device locale setting? That would avoid complex formatting logic trying to take account of different locale settings.

If that is done, then it should probably be implemented across Android and iOS. I will submit a separate issue for this suggestion in the [cwa-documentation](https://github.com/corona-warn-app/cwa-documentation/issues) repository.

**Comment by MikeMcC399** [comment]
For the general suggestion that all dates of birth from certificates should be displayed on Android and iOS in the format yyyy-mm-dd I have submitted https://github.com/corona-warn-app/cwa-documentation/issues/710.

**Comment by MikeMcC399** [comment]
The issue of incorrect data of birth formatting with locale set to English (United Kingdom) is resolved in CWA Android 2.20.3.

Moderators may like to add the `Fix 2.20` label to this issue.
